cmd/auth/auth.go

```diff
@@ -0,0 +1,200 @@
+
+	authenticationURL := supabaseAuthURL + "?" + queryParams.Encode()
+
+	server := &http.Server{Addr: authCallbackAddr}
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		defer shutdown(server)
+
+		code := r.URL.Query().Get("code")
+		if code == "" {
+			http.Error(w, "Error: auth-code query param not found", http.StatusBadRequest)
+			return
+		}
+
+		sessionData, err := getSession(code, codeVerifier)
+		if err != nil {
+			http.Error(w, "Error: Access token exchange failed", http.StatusInternalServerError)
+			return
+		}
+
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			fmt.Println("Error getting home directory:", err)
+			return
+		}
+
+		dirName := path.Join(homeDir, ".pizza")
+		if err := os.MkdirAll(dirName, os.ModePerm); err != nil {
+			http.Error(w, ".pizza directory couldn't be created", http.StatusInternalServerError)
+			return
+		}
+
+		jsonData, err := json.Marshal(sessionData)
+		if err != nil {
+			http.Error(w, "Error marshaling session data", http.StatusInternalServerError)
+			return
+		}
+
+		filePath := path.Join(dirName, sessionFileName)
+		if err := os.WriteFile(filePath, jsonData, 0o600); err != nil {
+			http.Error(w, "Error writing to file", http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "text/html; charset=utf-8")
+		_, err = w.Write([]byte(successHTML))
+		if err != nil {
+			fmt.Println("Error writing response:", err)
+			return
+		}
+
+		username := sessionData["user"].(map[string]interface{})["user_metadata"].(map[string]interface{})["user_name"].(string)
+		fmt.Println("🎉 Login successful 🎉")
+		fmt.Println("Welcome aboard", username, "🍕")
+
+	})
+
+	err = browser.OpenURL(authenticationURL)
+	if err != nil {
+		fmt.Println("Failed to open the browser 🤦‍♂️")
+		fmt.Println("Navigate to the following URL to begin authentication:")
+		fmt.Println(authenticationURL)
+	}
+
+	errCh := make(chan error)
+	go func() {
+		errCh <- server.ListenAndServe()
+	}()
+
+	interruptCh := make(chan os.Signal, 1)
+	signal.Notify(interruptCh, os.Interrupt)
+
+	select {
+	case err := <-errCh:
+		if err != nil && err != http.ErrServerClosed {
+			return err
+		}
+	case <-time.After(60 * time.Second):
+		shutdown(server)
+		return errors.New("authentication timeout")
+	case <-interruptCh:
+		fmt.Println("\nAuthentication interrupted❗️")
+		shutdown(server)
+		os.Exit(0)
+	}
+	return nil
+}
+
+func getSession(authCode, codeVerifier string) (map[string]interface{}, error) {
+	url := fmt.Sprintf("https://%s.supabase.co/auth/v1/token?grant_type=pkce", supabaseID)
+
+	payload := map[string]string{
+		"auth_code":     authCode,
+		"code_verifier": codeVerifier,
+	}
+
+	jsonPayload, _ := json.Marshal(payload)
+
+	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(jsonPayload))
+	req.Header.Set("Content-Type", "application/json;charset=UTF-8")
+	req.Header.Set("ApiKey", supabasePublicKey)
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
```

Comment:
These should be wrapped in `fmt.Errorf` so error tracability is easier. Something like:

```go
fmt.Errorf("could not make req with default client: %s", err.Error())
```